app/V1Module/presenters/GroupsPresenter.php

```diff
@@ -436,6 +468,18 @@ public function actionExercises(string $id) {
     $this->sendSuccessResponse($exercises);
   }
 
+  public function checkStats(string $id) {
+    $group = $this->groups->findOrThrow($id);
+
+    if (!$this->groupAcl->canViewStats($group)) {
+      $user = $this->getCurrentUser();
+
+      if (!($this->groupAcl->canViewStudentStats($group, $user) && $group->isStudentOf($user))) {
```

Comment:
Shouldn't this be negation of condition below on line 451?

Reply:
The method should check that the user either a) can view all statistics or b) is a student of the group and can view their own statistics. Is there any discrepancy between this and the code?

Reply:
Sorrry... my bad... I did not see parenthesis